From a8cde823805ab36d7d20e363f84eb6e12da9e543 Mon Sep 17 00:00:00 2001
From: effectsbot <eff3ctsb0t@gmail.com>
Date: Mon, 12 Jan 2026 14:30:53 +0100
Subject: [PATCH] Switch default FNAPlatform back to SDL2

Terraria still depends on SDL2, so I'll keep this platform as the default until further notice.
---
 src/FNAPlatform/FNAPlatform.cs | 132 ++++++++++++++++-----------------
 1 file changed, 66 insertions(+), 66 deletions(-)

diff --git a/src/FNAPlatform/FNAPlatform.cs b/src/FNAPlatform/FNAPlatform.cs
index 44e25f9..75fff5b 100644
--- a/src/FNAPlatform/FNAPlatform.cs
+++ b/src/FNAPlatform/FNAPlatform.cs
@@ -36,15 +36,15 @@ namespace Microsoft.Xna.Framework
 			 * -flibit
 			 */
 
-			bool useSDL2 = Environment.GetEnvironmentVariable("FNA_PLATFORM_BACKEND") == "SDL2";
+			bool useSDL3 = Environment.GetEnvironmentVariable("FNA_PLATFORM_BACKEND") == "SDL3";
 
-			if (!useSDL2)
+			if (!useSDL3)
 			{
-				SetEnv = SDL3_FNAPlatform.SetEnv;
+				SetEnv = SDL2_FNAPlatform.SetEnv;
 			}
 			else
 			{
-				SetEnv = SDL2_FNAPlatform.SetEnv;
+				SetEnv = SDL3_FNAPlatform.SetEnv;
 			}
 
 			// Built-in command line arguments
@@ -113,63 +113,7 @@ namespace Microsoft.Xna.Framework
 				Environment.GetEnvironmentVariable("FNA_SDL2_FORCE_BASE_PATH")
 			);
 
-			if (!useSDL2)
-			{
-				Malloc =			SDL3_FNAPlatform.Malloc;
-				Free =				SDL3.SDL.SDL_free;
-				CreateWindow =			SDL3_FNAPlatform.CreateWindow;
-				DisposeWindow =			SDL3_FNAPlatform.DisposeWindow;
-				ApplyWindowChanges =		SDL3_FNAPlatform.ApplyWindowChanges;
-				ScaleForWindow =		SDL3_FNAPlatform.ScaleForWindow;
-				GetWindowBounds =		SDL3_FNAPlatform.GetWindowBounds;
-				GetWindowResizable =		SDL3_FNAPlatform.GetWindowResizable;
-				SetWindowResizable =		SDL3_FNAPlatform.SetWindowResizable;
-				GetWindowBorderless =		SDL3_FNAPlatform.GetWindowBorderless;
-				SetWindowBorderless =		SDL3_FNAPlatform.SetWindowBorderless;
-				SetWindowTitle =		SDL3_FNAPlatform.SetWindowTitle;
-				IsScreenKeyboardShown =		SDL3_FNAPlatform.IsScreenKeyboardShown;
-				RegisterGame =			SDL3_FNAPlatform.RegisterGame;
-				UnregisterGame =		SDL3_FNAPlatform.UnregisterGame;
-				PollEvents =			SDL3_FNAPlatform.PollEvents;
-				GetGraphicsAdapters =		SDL3_FNAPlatform.GetGraphicsAdapters;
-				GetCurrentDisplayMode =		SDL3_FNAPlatform.GetCurrentDisplayMode;
-				GetMonitorHandle =		SDL3_FNAPlatform.GetMonitorHandle;
-				GetKeyFromScancode =		SDL3_FNAPlatform.GetKeyFromScancode;
-				IsTextInputActive =		SDL3_FNAPlatform.IsTextInputActive;
-				StartTextInput =		SDL3_FNAPlatform.StartTextInput;
-				StopTextInput =			SDL3_FNAPlatform.StopTextInput;
-				SetTextInputRectangle =		SDL3_FNAPlatform.SetTextInputRectangle;
-				GetMouseState =			SDL3_FNAPlatform.GetMouseState;
-				SetMousePosition =		SDL3_FNAPlatform.WarpMouseInWindow;
-				OnIsMouseVisibleChanged =	SDL3_FNAPlatform.OnIsMouseVisibleChanged;
-				GetRelativeMouseMode =		SDL3_FNAPlatform.GetRelativeMouseMode;
-				SetRelativeMouseMode =		SDL3_FNAPlatform.SetRelativeMouseMode;
-				GetGamePadCapabilities =	SDL3_FNAPlatform.GetGamePadCapabilities;
-				GetGamePadState =		SDL3_FNAPlatform.GetGamePadState;
-				SetGamePadVibration =		SDL3_FNAPlatform.SetGamePadVibration;
-				SetGamePadTriggerVibration =	SDL3_FNAPlatform.SetGamePadTriggerVibration;
-				GetGamePadGUID =		SDL3_FNAPlatform.GetGamePadGUID;
-				SetGamePadLightBar =		SDL3_FNAPlatform.SetGamePadLightBar;
-				GetGamePadGyro = 		SDL3_FNAPlatform.GetGamePadGyro;
-				GetGamePadAccelerometer =	SDL3_FNAPlatform.GetGamePadAccelerometer;
-				GetStorageRoot =		SDL3_FNAPlatform.GetStorageRoot;
-				GetDriveInfo =			SDL3_FNAPlatform.GetDriveInfo;
-				ReadFileToPointer =		SDL3_FNAPlatform.ReadToPointer;
-				FreeFilePointer =		SDL3_FNAPlatform.FreeFilePointer;
-				ShowRuntimeError =		SDL3_FNAPlatform.ShowRuntimeError;
-				GetMicrophones =		SDL3_FNAPlatform.GetMicrophones;
-				GetMicrophoneSamples =		SDL3_FNAPlatform.GetMicrophoneSamples;
-				GetMicrophoneQueuedBytes =	SDL3_FNAPlatform.GetMicrophoneQueuedBytes;
-				StartMicrophone =		SDL3_FNAPlatform.StartMicrophone;
-				StopMicrophone =		SDL3_FNAPlatform.StopMicrophone;
-				GetTouchCapabilities =		SDL3_FNAPlatform.GetTouchCapabilities;
-				UpdateTouchPanelState =		SDL3_FNAPlatform.UpdateTouchPanelState;
-				GetNumTouchFingers =		SDL3_FNAPlatform.GetNumTouchFingers;
-				SupportsOrientationChanges =	SDL3_FNAPlatform.SupportsOrientationChanges;
-				NeedsPlatformMainLoop = 	SDL3_FNAPlatform.NeedsPlatformMainLoop;
-				RunPlatformMainLoop =		SDL3_FNAPlatform.RunPlatformMainLoop;
-			}
-			else
+			if (!useSDL3)
 			{
 				Malloc =			SDL2_FNAPlatform.Malloc;
 				Free =				SDL2.SDL.SDL_free;
@@ -225,18 +169,74 @@ namespace Microsoft.Xna.Framework
 				NeedsPlatformMainLoop = 	SDL2_FNAPlatform.NeedsPlatformMainLoop;
 				RunPlatformMainLoop =		SDL2_FNAPlatform.RunPlatformMainLoop;
 			}
+			else
+			{
+				Malloc =			SDL3_FNAPlatform.Malloc;
+				Free =				SDL3.SDL.SDL_free;
+				CreateWindow =			SDL3_FNAPlatform.CreateWindow;
+				DisposeWindow =			SDL3_FNAPlatform.DisposeWindow;
+				ApplyWindowChanges =		SDL3_FNAPlatform.ApplyWindowChanges;
+				ScaleForWindow =		SDL3_FNAPlatform.ScaleForWindow;
+				GetWindowBounds =		SDL3_FNAPlatform.GetWindowBounds;
+				GetWindowResizable =		SDL3_FNAPlatform.GetWindowResizable;
+				SetWindowResizable =		SDL3_FNAPlatform.SetWindowResizable;
+				GetWindowBorderless =		SDL3_FNAPlatform.GetWindowBorderless;
+				SetWindowBorderless =		SDL3_FNAPlatform.SetWindowBorderless;
+				SetWindowTitle =		SDL3_FNAPlatform.SetWindowTitle;
+				IsScreenKeyboardShown =		SDL3_FNAPlatform.IsScreenKeyboardShown;
+				RegisterGame =			SDL3_FNAPlatform.RegisterGame;
+				UnregisterGame =		SDL3_FNAPlatform.UnregisterGame;
+				PollEvents =			SDL3_FNAPlatform.PollEvents;
+				GetGraphicsAdapters =		SDL3_FNAPlatform.GetGraphicsAdapters;
+				GetCurrentDisplayMode =		SDL3_FNAPlatform.GetCurrentDisplayMode;
+				GetMonitorHandle =		SDL3_FNAPlatform.GetMonitorHandle;
+				GetKeyFromScancode =		SDL3_FNAPlatform.GetKeyFromScancode;
+				IsTextInputActive =		SDL3_FNAPlatform.IsTextInputActive;
+				StartTextInput =		SDL3_FNAPlatform.StartTextInput;
+				StopTextInput =			SDL3_FNAPlatform.StopTextInput;
+				SetTextInputRectangle =		SDL3_FNAPlatform.SetTextInputRectangle;
+				GetMouseState =			SDL3_FNAPlatform.GetMouseState;
+				SetMousePosition =		SDL3_FNAPlatform.WarpMouseInWindow;
+				OnIsMouseVisibleChanged =	SDL3_FNAPlatform.OnIsMouseVisibleChanged;
+				GetRelativeMouseMode =		SDL3_FNAPlatform.GetRelativeMouseMode;
+				SetRelativeMouseMode =		SDL3_FNAPlatform.SetRelativeMouseMode;
+				GetGamePadCapabilities =	SDL3_FNAPlatform.GetGamePadCapabilities;
+				GetGamePadState =		SDL3_FNAPlatform.GetGamePadState;
+				SetGamePadVibration =		SDL3_FNAPlatform.SetGamePadVibration;
+				SetGamePadTriggerVibration =	SDL3_FNAPlatform.SetGamePadTriggerVibration;
+				GetGamePadGUID =		SDL3_FNAPlatform.GetGamePadGUID;
+				SetGamePadLightBar =		SDL3_FNAPlatform.SetGamePadLightBar;
+				GetGamePadGyro = 		SDL3_FNAPlatform.GetGamePadGyro;
+				GetGamePadAccelerometer =	SDL3_FNAPlatform.GetGamePadAccelerometer;
+				GetStorageRoot =		SDL3_FNAPlatform.GetStorageRoot;
+				GetDriveInfo =			SDL3_FNAPlatform.GetDriveInfo;
+				ReadFileToPointer =		SDL3_FNAPlatform.ReadToPointer;
+				FreeFilePointer =		SDL3_FNAPlatform.FreeFilePointer;
+				ShowRuntimeError =		SDL3_FNAPlatform.ShowRuntimeError;
+				GetMicrophones =		SDL3_FNAPlatform.GetMicrophones;
+				GetMicrophoneSamples =		SDL3_FNAPlatform.GetMicrophoneSamples;
+				GetMicrophoneQueuedBytes =	SDL3_FNAPlatform.GetMicrophoneQueuedBytes;
+				StartMicrophone =		SDL3_FNAPlatform.StartMicrophone;
+				StopMicrophone =		SDL3_FNAPlatform.StopMicrophone;
+				GetTouchCapabilities =		SDL3_FNAPlatform.GetTouchCapabilities;
+				UpdateTouchPanelState =		SDL3_FNAPlatform.UpdateTouchPanelState;
+				GetNumTouchFingers =		SDL3_FNAPlatform.GetNumTouchFingers;
+				SupportsOrientationChanges =	SDL3_FNAPlatform.SupportsOrientationChanges;
+				NeedsPlatformMainLoop = 	SDL3_FNAPlatform.NeedsPlatformMainLoop;
+				RunPlatformMainLoop =		SDL3_FNAPlatform.RunPlatformMainLoop;
+			}
 
 			FNALoggerEXT.Initialize();
 
-			if (!useSDL2)
+			if (!useSDL3)
 			{
-				AppDomain.CurrentDomain.ProcessExit += SDL3_FNAPlatform.ProgramExit;
-				TitleLocation = SDL3_FNAPlatform.ProgramInit(args);
+				AppDomain.CurrentDomain.ProcessExit += SDL2_FNAPlatform.ProgramExit;
+				TitleLocation = SDL2_FNAPlatform.ProgramInit(args);
 			}
 			else
 			{
-				AppDomain.CurrentDomain.ProcessExit += SDL2_FNAPlatform.ProgramExit;
-				TitleLocation = SDL2_FNAPlatform.ProgramInit(args);
+				AppDomain.CurrentDomain.ProcessExit += SDL3_FNAPlatform.ProgramExit;
+				TitleLocation = SDL3_FNAPlatform.ProgramInit(args);
 			}
 
 			/* Do this AFTER ProgramInit so the platform library
-- 
2.52.0

